CMakeLists.txt

```diff
@@ -292,6 +294,15 @@ if(openPMD_HAVE_PYTHON)
         INTERFACE pybind11::pybind11)
 endif()
 
+# external library: spdlog (optional)
+if(openPMD_USE_LOGGING STREQUAL AUTO)
+    set(openPMD_HAVE_LOGGING (${CMAKE_BUILD_TYPE} STREQUAL "Debug" OR ${CMAKE_BUILD_TYPE} STREQUAL "RelWithDebInfo"))
```

Comment:
We can do the control this way, but it's slightly different as the other dependencies then.

For example, if the `openPMD_USE_XY` is `AUTO` and set to `EXTERNAL` and *not* found, this will not fail in our current logic. Here it would.

Reply:
Shamelessly copied from Pybind11 as that probably is the most consistent with the other CMake `_USE_` logic.

Reply:
Yes, I just wonder if we should couple it to the Debug build logic since we will use log levels on various intensity also in Release builds and vice versa, might want to disable it for some Debug runs.

So maybe just handle it's default activation just like any other functional extension? Just a thought.